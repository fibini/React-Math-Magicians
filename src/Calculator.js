import { useState } from 'react';
import calculate from './logic/calculate';

const Calculator = () => {
  const [emptycalc, setcalc] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setcalc(calculate(emptycalc, e.target.innerHTML));
  };
  return (
    <ul className="calculator">
      <li data-testid="answer-slot" className="answer">{emptycalc.next || emptycalc.total || '0'}</li>
      <button
        type="button"
        name="AC"
        className="ac lightgrey"
        onClick={handleClick}
      >
        AC
      </button>
      <button
        type="button"
        name="+/-"
        className="plus-minus lightgrey"
        onClick={handleClick}
      >
        +/-
      </button>
      <button
        type="button"
        name="%"
        className="percentage lightgrey"
        onClick={handleClick}
      >
        %
      </button>
      <button
        type="button"
        name="7"
        className="seven lightgrey"
        onClick={handleClick}
      >
        7
      </button>
      <button
        data-testid="eight"
        type="button"
        name="8"
        className="eight lightgrey"
        onClick={handleClick}
      >
        8
      </button>
      <button
        type="button"
        name="รท"
        className="division orange"
        onClick={handleClick}
      >
        รท
      </button>
      <button
        type="button"
        name="9"
        className="nine lightgrey"
        onClick={handleClick}
      >
        9
      </button>
      <button
        type="button"
        name="x"
        className="multiply orange"
        onClick={handleClick}
      >
        x
      </button>
      <button
        type="button"
        name="4"
        className="four lightgrey"
        onClick={handleClick}
      >
        4
      </button>
      <button
        type="button"
        name="5"
        className="five lightgrey"
        onClick={handleClick}
      >
        5
      </button>
      <button
        type="button"
        name="6"
        className="six lightgrey"
        onClick={handleClick}
      >
        6
      </button>
      <button
        type="button"
        name="-"
        className="minus orange"
        onClick={handleClick}
      >
        -
      </button>
      <button
        type="button"
        name="1"
        className="one lightgrey"
        onClick={handleClick}
      >
        1
      </button>
      <button
        type="button"
        name="2"
        className="two lightgrey"
        onClick={handleClick}
      >
        2
      </button>
      <button
        type="button"
        name="3"
        className="three lightgrey"
        onClick={handleClick}
      >
        3
      </button>
      <button
        type="button"
        name="+"
        className="plus orange"
        onClick={handleClick}
      >
        +
      </button>
      <button
        type="button"
        name="0"
        className="zero lightgrey"
        onClick={handleClick}
      >
        0
      </button>
      <button
        type="button"
        name="."
        className="decimal lightgrey"
        onClick={handleClick}
      >
        .
      </button>
      <button
        type="button"
        name="="
        className="equal orange"
        onClick={handleClick}
      >
        =
      </button>
    </ul>
  );
};

export default Calculator;
