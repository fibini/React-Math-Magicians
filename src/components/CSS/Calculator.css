.calculator {
  display: grid;
  justify-items: center;
  direction: rtl;
  background-color: blue;
  margin-top: 10%;
  margin-left: 10%;
  width: 280px;
  border-radius: 10%;
  grid-template-areas:
    "answer answer answer answer"
    "division percentage plus-minus ac"
    "multiply nine eight seven"
    "minus six five four"
    "plus three two one"
    "equal decimal zero zero";
}

.answer {
  grid-area: answer;
  color: white;
  font-size: 35px;
  width: 280px;
  background-color: #858693;
  border-top: 2px solid lightgrey;
  border-left: 2px solid lightgrey;
  border-right: 2px solid lightgrey;
  line-height: 250%;
  border-radius: 10%;
}

.ac {
  grid-area: ac;
  width: 100%;
}

.plus-minus {
  direction: ltr;
  grid-area: plus-minus;
  width: 100%;
}

.percentage {
  grid-area: percentage;
  width: 100%;
}

.division {
  grid-area: division;
  width: 100%;
}

.seven {
  grid-area: seven;
  width: 100%;
}

.eight {
  grid-area: eight;
  width: 100%;
}

.nine {
  grid-area: nine;
  width: 100%;
}

.multiply {
  grid-area: multiply;
  width: 100%;
}

.four {
  grid-area: four;
  width: 100%;
}

.five {
  grid-area: five;
  width: 100%;
}

.six {
  grid-area: six;
  width: 100%;
}

.minus {
  grid-area: minus;
  width: 100%;
}

.one {
  grid-area: one;
  width: 100%;
}

.two {
  grid-area: two;
}

.three {
  grid-area: three;
  width: 100%;
}

.plus {
  grid-area: plus;
  width: 100%;
}

.zero {
  grid-area: zero;
  width: 100%;
}

.decimal {
  grid-area: decimal;
  width: 100%;
}

.equal {
  grid-area: equal;
  width: 100%;
}

.lightgrey {
  cursor: pointer;
  line-height: 400%;
  font-weight: bold;
  border-top: 2px solid lightgrey;
  border-left: 2px solid lightgrey;
  text-align: center;
  background-color: #e0e0e0;
  border-radius: 10%;
}

.orange {
  cursor: pointer;
  line-height: 400%;
  font-weight: bold;
  border-top: 2px solid lightgrey;
  border-left: 2px solid lightgrey;
  border-right: 2px solid lightgrey;
  text-align: center;
  background-color: #f5913e;
  border-radius: 10%;
}

.width {
  width: 70px;
}
