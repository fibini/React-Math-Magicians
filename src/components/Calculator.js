import { useState } from 'react';
import './CSS/Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [emptycalc, setcalc] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    setcalc(calculate(emptycalc, e.target.innerHTML));
  };
  return (
    <ul className="calculator">
      <li data-testid="answer-slot" className="answer">{emptycalc.next || emptycalc.total || '0'}</li>
      <button
        type="button"
        name="AC"
        className="ac lightgrey width"
        onClick={handleClick}
      >
        AC
      </button>
      <button
        type="button"
        name="+/-"
        className="plus-minus lightgrey width"
        onClick={handleClick}
      >
        +/-
      </button>
      <button
        type="button"
        name="%"
        className="percentage lightgrey width"
        onClick={handleClick}
      >
        %
      </button>
      <button
        type="button"
        name="7"
        className="seven lightgrey width"
        onClick={handleClick}
      >
        7
      </button>
      <button
        data-testid="eight"
        type="button"
        name="8"
        className="eight lightgrey width"
        onClick={handleClick}
      >
        8
      </button>
      <button
        type="button"
        name="รท"
        className="division orange width"
        onClick={handleClick}
      >
        รท
      </button>
      <button
        type="button"
        name="9"
        className="nine lightgrey width"
        onClick={handleClick}
      >
        9
      </button>
      <button
        type="button"
        name="x"
        className="multiply orange width"
        onClick={handleClick}
      >
        x
      </button>
      <button
        type="button"
        name="4"
        className="four lightgrey width"
        onClick={handleClick}
      >
        4
      </button>
      <button
        type="button"
        name="5"
        className="five lightgrey width"
        onClick={handleClick}
      >
        5
      </button>
      <button
        type="button"
        name="6"
        className="six lightgrey width"
        onClick={handleClick}
      >
        6
      </button>
      <button
        type="button"
        name="-"
        className="minus orange width"
        onClick={handleClick}
      >
        -
      </button>
      <button
        type="button"
        name="1"
        className="one lightgrey width"
        onClick={handleClick}
      >
        1
      </button>
      <button
        type="button"
        name="2"
        className="two lightgrey width"
        onClick={handleClick}
      >
        2
      </button>
      <button
        type="button"
        name="3"
        className="three lightgrey width"
        onClick={handleClick}
      >
        3
      </button>
      <button
        type="button"
        name="+"
        className="plus orange width"
        onClick={handleClick}
      >
        +
      </button>
      <button
        type="button"
        name="0"
        className="zero lightgrey"
        onClick={handleClick}
      >
        0
      </button>
      <button
        type="button"
        name="."
        className="decimal lightgrey width"
        onClick={handleClick}
      >
        .
      </button>
      <button
        type="button"
        name="="
        className="equal orange width"
        onClick={handleClick}
      >
        =
      </button>
    </ul>
  );
};

export default Calculator;
