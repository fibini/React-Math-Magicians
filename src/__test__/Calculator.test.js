import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Calculator from '../Calculator';
import '@testing-library/jest-dom/extend-expect';

test('Calculator correct click functionality', () => {
  const { getByTestId } = render(<Calculator />);
  const number = getByTestId('eight');
  const slot = getByTestId('answer-slot');
  fireEvent.click(number);
  expect(slot.childNodes[0].data).toBe('8');
});
