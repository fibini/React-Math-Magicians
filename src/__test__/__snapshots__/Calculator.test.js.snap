// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<ul
  className="calculator"
>
  <li
    className="answer"
    data-testid="answer-slot"
  >
    0
  </li>
  <button
    className="ac lightgrey"
    name="AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="plus-minus lightgrey"
    name="+/-"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="percentage lightgrey"
    name="%"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="seven lightgrey"
    name="7"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="eight lightgrey"
    data-testid="eight"
    name="8"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="division orange"
    name="รท"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="nine lightgrey"
    name="9"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="multiply orange"
    name="x"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="four lightgrey"
    name="4"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="five lightgrey"
    name="5"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="six lightgrey"
    name="6"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="minus orange"
    name="-"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="one lightgrey"
    name="1"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="two lightgrey"
    name="2"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="three lightgrey"
    name="3"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="plus orange"
    name="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="zero lightgrey"
    name="0"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="decimal lightgrey"
    name="."
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="equal orange"
    name="="
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</ul>
`;
